model-optimizer/mo/utils/versions_checker.py

```diff
@@ -82,14 +86,28 @@ def parse_and_filter_versions_list(required_fw_versions, version_list, env_setup
                 # this python_version requirement is not satisfied to required environment
                 # and requirement for a dependency will be skipped
                 return version_list
+        elif env_marker == 'sys_platform' and env_marker in env_setup:
+            splited_env_req[1] = splited_env_req[1].strip(' ').replace('\'', '')
+            if '==' in env_req:
+                if not env_setup['sys_platform'] == splited_env_req[1]:
+                    # this sys_platform requirement is not satisfied to required environment
+                    # and requirement for a dependency will be skipped
+                    return version_list
+            elif '!=' in env_req:
+                if not env_setup['sys_platform'] != splited_env_req[1]:
+                    # this sys_platform requirement is not satisfied to required environment
+                    # and requirement for a dependency will be skipped
+                    return version_list
```

Comment:
your implementation does not support a case with several marker values comparison, for example,
sys_platform!=win_32,!=darwin. You need to like it is done for `python_version` and it seems these two branches can be merged.